torch/nn/functional.py

```diff
@@ -1178,6 +1178,30 @@ def local_response_norm(input, size, alpha=1e-4, beta=0.75, k=1):
     return input / div
 
 
+def layer_norm(input, weight=None, bias=None, eps=1e-5):
+    if input is not None and input.dim() != 2:
+        raise ValueError("Expected 2D tensor as input, got {}D tensor instead.".format(input.dim()))
+
+    mean = input.mean(1, keepdim=True)
+    # Prevent NaN gradients when sample std is 0 by using alternative standard deviation calculation
+    std = ((input - mean).pow(2).sum(1, keepdim=True).div(input.size(1) - 1) + eps).sqrt()
```

Comment:
How does this help?

Reply:
@mys007 can you explain this one?

Reply:
Kaixhin's commit was the reaction to my report here: https://github.com/pytorch/pytorch/pull/2019#issuecomment-342787832 . But maybe the `std()` implementation was fixed in the meantime?